website/docs/configuration/resources.html.md

```diff
@@ -445,8 +445,10 @@ can't refer to any resource attributes that aren't known until after a
 configuration is applied (such as a unique ID generated by the remote API when
 an object is created).
 
-The `for_each` value must be a map or set with one element per desired
-resource instance. If you need to declare resource instances based on a nested
+The `for_each` value must be an explicit map or set with one element per desired
+resource instance. To prevent unwanted surprises during conversion, the `for_each` 
+argument does not support implicit type conversion.
+If you need to declare resource instances based on a nested
```

Comment:
Hmm, this is a very good point to call out 👍🏼, but there's a few issues to iron out first: 

- We don't really define what an "explicit map" or "explicit set" is anywhere else in the docs, so introducing that term here is somewhat confusing. 
- Turns out, it's only sets that have the issue, not maps. The familiar "map" literal syntax in Terraform is actually an _object_ literal, but we don't need to use the `tomap` function on it because `for_each` doesn't mind converting those.

Taking just a quick stab at this, how about something like: 

```suggestion
The `for_each` value must be a map or set with one element per desired
resource instance. When providing a set, you must use an expression that 
explicitly returns a set value, like the [`toset`](/docs/configuration/functions/toset.html)
function; to prevent unwanted surprises during conversion, the `for_each` 
argument does not implicitly convert lists or tuples to sets.
If you need to declare resource instances based on a nested
```

Or if you've got another idea about how to word that.

Reply:
* Agreed on avoiding 'explicit map/set'. 
* It does convert objects? Remarkable.
I like the suggestion! I'll commit it.